src/um/dwrite_1.rs

```diff
@@ -0,0 +1,724 @@
+}}
+
+ENUM!{ enum DWRITE_VERTICAL_GLYPH_ORIENTATION {
+    DWRITE_VERTICAL_GLYPH_ORIENTATION_DEFAULT = 0x0, // 0
+    DWRITE_VERTICAL_GLYPH_ORIENTATION_STACKED = 0x1, // 1
+}}
+
+ENUM!{ enum DWRITE_GLYPH_ORIENTATION_ANGLE {
+    DWRITE_GLYPH_ORIENTATION_ANGLE_0_DEGREES = 0x0, // 0
+    DWRITE_GLYPH_ORIENTATION_ANGLE_90_DEGREES = 0x1, // 1
+    DWRITE_GLYPH_ORIENTATION_ANGLE_180_DEGREES = 0x2, // 2
+    DWRITE_GLYPH_ORIENTATION_ANGLE_270_DEGREES = 0x3, // 3
+}}
+
+STRUCT!{struct DWRITE_FONT_METRICS1 {
+    designUnitsPerEm: UINT16,
+    ascent: UINT16,
+    descent: UINT16,
+    lineGap: INT16,
+    capHeight: UINT16,
+    xHeight: UINT16,
+    underlinePosition: INT16,
+    underlineThickness: UINT16,
+    strikethroughPosition: INT16,
+    strikethroughThickness: UINT16,
+    glyphBoxLeft: INT16,
+    glyphBoxTop: INT16,
+    glyphBoxRight: INT16,
+    glyphBoxBottom: INT16,
+    subscriptPositionX: INT16,
+    subscriptPositionY: INT16,
+    subscriptSizeX: INT16,
+    subscriptSizeY: INT16,
+    superscriptPositionX: INT16,
+    superscriptPositionY: INT16,
+    superscriptSizeX: INT16,
+    superscriptSizeY: INT16,
+    hasTypographicMetrics: BOOL,
+}}
+
+STRUCT!{struct DWRITE_CARET_METRICS {
+    slopeRise: INT16,
+    slopeRun: INT16,
+    offset: INT16,
+}}
+
+STRUCT!{struct DWRITE_PANOSE_text {
+    familyKind: UINT8,
+    serifStyle: UINT8,
+    weight: UINT8,
+    proportion: UINT8,
+    contrast: UINT8,
+    strokeVariation: UINT8,
+    armStyle: UINT8,
+    letterform: UINT8,
+    midline: UINT8,
+    xHeight: UINT8,
+}}
+
+STRUCT!{struct DWRITE_PANOSE_script {
+    familyKind: UINT8,
+    toolKind: UINT8,
+    weight: UINT8,
+    spacing: UINT8,
+    aspectRatio: UINT8,
+    contrast: UINT8,
+    scriptTopology: UINT8,
+    scriptForm: UINT8,
+    finials: UINT8,
+    xAscent: UINT8,
+}}
+
+STRUCT!{struct DWRITE_PANOSE_decorative {
+    familyKind: UINT8,
+    decorativeClass: UINT8,
+    weight: UINT8,
+    aspect: UINT8,
+    contrast: UINT8,
+    serifVariant: UINT8,
+    fill: UINT8,
+    lining: UINT8,
+    decorativeTopology: UINT8,
+    characterRange: UINT8,
+}}
+
+STRUCT!{struct DWRITE_PANOSE_symbol {
+    familyKind: UINT8,
+    symbolKind: UINT8,
+    weight: UINT8,
+    spacing: UINT8,
+    aspectRatioAndContrast: UINT8,
+    aspectRatio94: UINT8,
+    aspectRatio119: UINT8,
+    aspectRatio157: UINT8,
+    aspectRatio163: UINT8,
+    aspectRatio211: UINT8,
+}}
+
+STRUCT!{struct DWRITE_PANOSE {
+    u1: [UINT8; 10],
+}}
+
+UNION!(DWRITE_PANOSE, u1, values, values_mut, [UINT8; 10]);
+UNION!(DWRITE_PANOSE, u1, familyKind, familyKind_mut, UINT8);
+UNION!(DWRITE_PANOSE, u1, text, text_mut, DWRITE_PANOSE_text);
+UNION!(DWRITE_PANOSE, u1, script, script_mut, DWRITE_PANOSE_script);
+UNION!(DWRITE_PANOSE, u1, decorative, decorative_mut, DWRITE_PANOSE_decorative);
+UNION!(DWRITE_PANOSE, u1, symbol, symbol_mut, DWRITE_PANOSE_symbol);
+
+STRUCT!{struct DWRITE_UNICODE_RANGE {
+    first: UINT32,
+    last: UINT32,
+}}
+
+IFDEF!{
```

Comment:
You should only use `IFDEF!` when you're applying a `cfg` to a group of definitions.